### ISSUE TITLE
Write up all classes in Java (will have to explore plotting and dataframe methods)

### ISSUE THREAD

**Issue by mghosh00** [mghosh00/BasicNeuralNetwork#12]
Here is a list of packages that need to be created for this issue:

- [x] components
- [x] data_generators
- [x] functions
- [x] learning
- [x] util

**Comment by mghosh00** [comment]
## Subtasks
* components

- [ ] Edge
- [ ] Layer
- [ ] Network
- [ ] Neuron

* data_generators

- [ ] DataGenerator (abstract)
- [ ] NormalDataGenerator
- [ ] UniformDataGenerator

* functions

- [ ] Function (abstract)
- [ ] CrossEntropyLoss
- [ ] MSELoss
- [ ] ReLU
- [ ] Sigmoid
- [ ] Softmax
- [ ] Transfer

* learning

- [ ] Learner (abstract)
- [ ] Plotter
- [ ] Tester
- [ ] Trainer
- [ ] Validator

* util

- [ ] DataSplitter
- [ ] Partitioner
- [ ] WeightedPartitioner
